src/appMain/life_cycle_impl.h

```diff
@@ -0,0 +1,147 @@
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#ifndef SRC_APPMAIN_LIFE_CYCLE_IMPL_H_
+#define SRC_APPMAIN_LIFE_CYCLE_IMPL_H_
+
+#include "appMain/life_cycle.h"
+
+#include <unistd.h>
+#include <thread>
+#include <memory>
+#include "utils/macro.h"
+
+#include "config_profile/profile.h"
+#include "hmi_message_handler/hmi_message_handler_impl.h"
+#if (defined(MESSAGEBROKER_HMIADAPTER) || defined(PASA_HMI))
+#include "hmi_message_handler/messagebroker_adapter.h"
+#endif  // #if ( defined (MESSAGEBROKER_HMIADAPTER) || defined(PASA_HMI)  )
+#include "application_manager/application_manager_impl.h"
+#include "connection_handler/connection_handler_impl.h"
+#include "protocol_handler/protocol_handler_impl.h"
+#include "transport_manager/transport_manager.h"
+#include "transport_manager/transport_manager_default.h"
+#include "media_manager/media_manager_impl.h"
+#ifdef TELEMETRY_MONITOR
+#include "telemetry_monitor/telemetry_monitor.h"
+#endif
+
+#ifdef ENABLE_SECURITY
+namespace security_manager {
+class CryptoManager;
+class SecurityManagerImpl;
+}  // namespace security_manager
+#endif  // ENABLE_SECURITY
+
+namespace utils {
+class SystemTimeHandler;
+}  // namespace utils
+
+namespace main_namespace {
+
+class LowVoltageSignalsHandler;
+
+class LifeCycleImpl : public LifeCycle {
+ public:
+  explicit LifeCycleImpl(const profile::Profile& profile);
+
+  /**
+   * Creates and starts all system components
+   * @return true if all components started successfully
+   * otherwise false.
+   */
+  bool StartComponents() OVERRIDE;
+
+  /**
+   * Initialize MessageBroker component
+   * @return true if success otherwise false.
+   */
+  bool InitMessageSystem() OVERRIDE;
+  /**
+   * @brief Main loop
+   */
+  void Run() OVERRIDE;
+
+  /**
+   * Stops all system components
+   */
+  void StopComponents() OVERRIDE;
+
+  /**
+   * Makes appropriate actions when Low Voltage signal received:
+   * Stops all SDL activities except of waiting of UNIX signals
+   * from HMI
+   */
+  void LowVoltage() OVERRIDE;
+
+  /**
+   * Makes appropriate actions when Wake Up signal received:
+   * Restores all SDL activities stopped due to LOW VOLTAGE
+   * from HMI
+   */
+  void WakeUp() OVERRIDE;
+
+  /**
+   * Makes appropriate actions when Ignition Off signal received:
+   * Triggers all SDL components stop and deletion
+   */
+  void IgnitionOff() OVERRIDE;
+
+ private:
+  transport_manager::TransportManagerImpl* transport_manager_;
+  protocol_handler::ProtocolHandlerImpl* protocol_handler_;
+  connection_handler::ConnectionHandlerImpl* connection_handler_;
+  application_manager::ApplicationManagerImpl* app_manager_;
+  std::unique_ptr<LowVoltageSignalsHandler> low_voltage_signals_handler_;
```

Comment:
why this one is unique_ptr, but others are normal pointers?

Reply:
@yang1070 because `std::unique_ptr<LowVoltageSignalsHandler> low_voltage_signals_handler_;` is the new code and the other pointers are old code. Therefore I don't want to make changes unrelated to feature.